### ISSUE TITLE
An inconsistent detection result of  rule SA_FIELD_SELF_ASSIGNMENT

### ISSUE THREAD

**Issue by vanguard-1024** [spotbugs/spotbugs#2142]
Hi, I found an inconsistent detection result about the rule [SA_FIELD_SELF_ASSIGNMENT](https://spotbugs.readthedocs.io/en/latest/bugDescriptions.html#sa-self-assignment-of-field-sa-field-self-assignment), the code example is below. SpotBugs can detect the bug in line 4, but cannot in line 8. These two lines are equivalent, hence, I think this is a false negative.

```java
public class C {
    int foo;
    void foo1() {
        foo = foo++;  // can report a warning in this line
    }
    class subC {
        void foo2() {
            foo = foo++; // should report a warning in this line
        }
    }
}
```
SpotBugs version: 4.7.1
